File: Leetcode/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        unordered_map <int, int> A;
        vector <int> haveHalfOfTarget;

        for (size_t i = 0 ; i < nums.size(); ++i) {
            A[nums.at(i)] = static_cast<int>(i);
            if (nums.at(i) * 2 == target) {
                haveHalfOfTarget.emplace_back(i);
            }
        }

        if (haveHalfOfTarget.size() == 2) {
            return haveHalfOfTarget;
        }

        vector <int> res(2);
        for (auto it = A.begin(); it != A.end(); ++it) {
            if ( A.find(target - it->first) != A.end() &&  it->first * 2 != target ) {
                res.at(0) = it->second;
                res.at(1) = A[target - it->first];
                break;
            }
        }
        if (res.at(0) > res.at(1)) {
            swap(res.at(0), res.at(1));
        }
        return res;
    }
};

File: Leetcode/1/1.go
func twoSum(nums []int, target int) []int {
	var m map[int][]int
	var res []int
	m = make(map[int][]int)
	for i, num := range nums {
		m[num] = append(m[num], i)
	}
	if target % 2 == 0 {
		val, ok := m[target/2]
		if ok && len(val) > 1 {
			res = append(res, val[0])
			res = append(res, val[1])
			return res
		}
	}
	for k, v := range m {
		val, ok := m[target - k]
		if target - k != k && ok {
			res = append(res, v[0])
			res = append(res, val[0])
			break
		}
	}
	return res
}